src/job.ts

```diff
@@ -190,41 +194,6 @@ class Job extends Operation {
        * });
        */
       get: true,
-
-      /**
-       * Get the metadata of the job. This will mostly be useful for checking
-       * the status of a previously-run job.
-       *
-       * @see [Jobs: get API Documentation]{@link https://cloud.google.com/bigquery/docs/reference/v2/jobs/get}
-       *
-       * @method Job#getMetadata
-       * @param {function} [callback] The callback function.
-       * @param {?error} callback.err An error returned while making this
-       *     request.
-       * @param {object} callback.metadata The metadata of the job.
-       * @param {object} callback.apiResponse The full API response.
-       * @returns {Promise}
-       *
-       * @example
-       * const {BigQuery} = require('@google-cloud/bigquery');
-       * const bigquery = new BigQuery();
-       *
-       * const job = bigquery.job('id');
-       * job.getMetadata((err, metadata, apiResponse) => {});
-       *
-       * //-
-       * // If the callback is omitted, we'll return a Promise.
-       * //-
-       * job.getMetadata().then((data) => {
-       *   const metadata = data[0];
-       *   const apiResponse = data[1];
-       * });
-       */
-      getMetadata: {
-        reqOpts: {
-          qs: {location},
```

Comment:
I wonder if a live getter would work here:

```js
getMetadata: {
  reqOpts: {
    qs: {
      get location() {
        return this.location;
      }
    }
  }
}
```

Reply:
@stephenplusplus The above given solution throws below Error becase of scoping issue:
 `RangeError: Maximum call stack size exceeded javascript`

I did try with below but it also doesn't work:

```ts
getMetadata: {
  reqOpts: {
    qs: () => {
      this.location;
    },
  },
},
```

Reply:
Woops, I made a diff of what should work:

```diff
diff --git a/src/job.ts b/src/job.ts
index ab25289..e12befb 100644
--- a/src/job.ts
+++ b/src/job.ts
@@ -126,10 +126,7 @@ class Job extends Operation {
     options?: QueryResultsOptions
   ) => ResourceStream<RowMetadata>;
   constructor(bigQuery: BigQuery, id: string, options?: JobOptions) {
-    let location;
-    if (options && options.location) {
-      location = options.location;
-    }
+    let location: string;
 
     const methods = {
       /**
@@ -222,7 +219,12 @@ class Job extends Operation {
        */
       getMetadata: {
         reqOpts: {
-          qs: {location},
+          qs: {
+            get location() {
+              console.log({location})
+              return location;
+            },
+          },
         },
       },
     };
@@ -234,6 +236,15 @@ class Job extends Operation {
       methods,
     });
 
+    Object.defineProperty(this, 'location', {
+      get() {
+        return location;
+      },
+      set(_location) {
+        location = _location;
+      },
+    });
+
     this.bigQuery = bigQuery;
 
     if (options && options.location) {
```

Reply:
Done Thanks!!